lib/pages/settings/view/ask_permissions.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:acs_upb_mobile/authentication/model/user.dart';
+import 'package:acs_upb_mobile/authentication/service/auth_provider.dart';
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:acs_upb_mobile/pages/settings/model/request.dart';
+import 'package:acs_upb_mobile/pages/settings/service/request_provider.dart';
+import 'package:acs_upb_mobile/widgets/scaffold.dart';
+import 'package:acs_upb_mobile/widgets/toast.dart';
+import 'package:provider/provider.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+class AskPermissions extends StatefulWidget {
+  static const String routeName = '/requestPermissions';
+
+  @override
+  State<StatefulWidget> createState() => AskPermissionsState();
+}
+
+class AskPermissionsState extends State<AskPermissions> {
+  User user;
+  String requestBody = '';
+  bool agreedToResponsibilities = false;
+
+  /*
+   * This flag is used whenever an user has already submitted a request in order
+   * to verify if he is actually wants to resubmit another request.
+   */
+  bool informedAboutExistingRequest = false;
+
+  _fetchUser() async {
+    AuthProvider authProvider = Provider.of(context, listen: false);
+    user = await authProvider.currentUser;
+    if (mounted) {
+      setState(() {});
+    }
+  }
+
+  initState() {
+    super.initState();
+    _fetchUser();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    RequestProvider requestProvider =
+        Provider.of<RequestProvider>(context, listen: false);
+
+    return AppScaffold(
+        title: S.of(context).navigationAskPermissions,
+        actions: [
+          AppScaffoldAction(
+              text: S.of(context).buttonSave,
+              onPressed: () async {
+                if (!agreedToResponsibilities) {
+                  AppToast.show(S.of(context).warningAgreeTo +
+                      S.of(context).labelPermissionsConsent +
+                      '.');
+                  return;
+                }
+
+                if (requestBody == '') {
+                  AppToast.show(S.of(context).warningRequestEmpty);
+                  return;
+                }
+
+                final Request myRequest = Request(user.uid, requestBody);
+
+                /*
+                 * Check if there is already a request registered for the current
+                 * user.
+                 */
+                var queryResult =
+                    await requestProvider.isAlreadyRequested(myRequest);
+
+                if (queryResult && !informedAboutExistingRequest) {
+                  AppToast.show(S.of(context).messageRequestAlreadyExists);
```

Comment:
This is extremely odd UX. Why would you ask the user to press the same button twice to confirm? It's very confusing.

I think I told you to *show a dialog* for the user to confirm overwriting the previous request. A toast is not a dialog. Check out the edit_profile_page.dart for some examples of dialogues.

Reply:
😝😝😝